The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9337307621/job/25699011464) [(plain)](https://github.com/rust-lang/rust/commit/37af7879705551ba7ae3b20ded7916e88d646ef2/checks/25699011464/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- library/core/src/num/mod.rs - num::i128::cast_unsigned (line 391) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:394:14
  |
6 | assert_eq!(n.cast_unsigned(), u128::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::i16::cast_unsigned (line 322) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:325:14
  |
6 | assert_eq!(n.cast_unsigned(), u16::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::i32::cast_unsigned (line 345) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:348:14
  |
6 | assert_eq!(n.cast_unsigned(), u32::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::i64::cast_unsigned (line 368) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:371:14
  |
6 | assert_eq!(n.cast_unsigned(), u64::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::i8::cast_unsigned (line 299) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:302:14
  |
6 | assert_eq!(n.cast_unsigned(), u8::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::isize::cast_unsigned (line 465) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:468:14
  |
6 | assert_eq!(n.cast_unsigned(), usize::MAX);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::u128::cast_signed (line 1203) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:1206:14
  |
6 | assert_eq!(n.cast_signed(), -1i128);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::u16::cast_signed (line 1106) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:1109:14
  |
6 | assert_eq!(n.cast_signed(), -1i16);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::u32::cast_signed (line 1155) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:1158:14
  |
6 | assert_eq!(n.cast_signed(), -1i32);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::u64::cast_signed (line 1179) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:1182:14
  |
6 | assert_eq!(n.cast_signed(), -1i64);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::u8::cast_signed (line 491) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:494:14
  |
6 | assert_eq!(n.cast_signed(), -1i8);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
---- library/core/src/num/mod.rs - num::usize::cast_signed (line 1279) stdout ----
error[E0658]: use of unstable library feature 'integer_sign_cast'
##[error] --> library/core/src/num/mod.rs:1282:14
  |
6 | assert_eq!(n.cast_signed(), -1isize);
  |
  = note: see issue #125882 <https://github.com/rust-lang/rust/issues/125882> for more information
  = help: add `#![feature(integer_sign_cast)]` to the crate attributes to enable
  = note: this compiler was built on 2024-05-25; consider upgrading it if it is out of date
---
    library/core/src/num/mod.rs - num::usize::cast_signed (line 1279)

test result: FAILED. 4632 passed; 12 failed; 42 ignored; 0 measured; 0 filtered out; finished in 146.23s

error: doctest failed, to rerun pass `-p core --doc`
  local time: Sun Jun  2 09:57:21 UTC 2024
  network time: Sun, 02 Jun 2024 09:57:21 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/num/int_macros.rs
+++ b/library/core/src/num/int_macros.rs
@@ -185,8 +185,8 @@ macro_rules! int_impl {
 
         /// Returns the bit pattern of `self` reinterpreted as an unsigned integer of the same size.
         ///
-        /// This is a bit safer than `as` because it wouldn't silently change the size if the code
-        /// is refactored.
+        /// This produces the same result as an `as` cast, but ensures that the bit-width remains
+        /// the same.
         ///
         /// # Examples
         ///

--- a/library/core/src/num/int_macros.rs
+++ b/library/core/src/num/int_macros.rs
@@ -183,6 +183,30 @@ macro_rules! int_impl {
             (self as $UnsignedT).trailing_ones()
         }
 
+        /// Returns the bit pattern of `self` reinterpreted as an unsigned integer of the same size.
+        ///
+        /// This is a bit safer than `as` because it wouldn't silently change the size if the code
+        /// is refactored.
+        ///
+        /// # Examples
+        ///
+        /// Basic usage:
+        ///
+        /// ```
+        /// #![feature(integer_sign_cast)]
+        ///
+        #[doc = concat!("let n = -1", stringify!($SelfT), ";")]
+        ///
+        #[doc = concat!("assert_eq!(n.cast_unsigned(), ", stringify!($UnsignedT), "::MAX);")]
+        /// ```
+        #[unstable(feature = "integer_sign_cast", issue = "125882")]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline(always)]
+        pub const fn cast_unsigned(self) -> $UnsignedT {
+            self as $UnsignedT
+        }
+
         /// Shifts the bits to the left by a specified amount, `n`,
         /// wrapping the truncated bits to the end of the resulting integer.
         ///

--- a/library/core/src/num/uint_macros.rs
+++ b/library/core/src/num/uint_macros.rs
@@ -186,8 +186,8 @@ macro_rules! uint_impl {
 
         /// Returns the bit pattern of `self` reinterpreted as a signed integer of the same size.
         ///
-        /// This is a bit safer than `as` because it wouldn't silently change the size if the code
-        /// is refactored.
+        /// This produces the same result as an `as` cast, but ensures that the bit-width remains
+        /// the same.
         ///
         /// # Examples
         ///

--- a/library/core/src/num/uint_macros.rs
+++ b/library/core/src/num/uint_macros.rs
@@ -184,6 +184,30 @@ macro_rules! uint_impl {
             (!self).trailing_zeros()
         }
 
+        /// Returns the bit pattern of `self` reinterpreted as a signed integer of the same size.
+        ///
+        /// This is a bit safer than `as` because it wouldn't silently change the size if the code
+        /// is refactored.
+        ///
+        /// # Examples
+        ///
+        /// Basic usage:
+        ///
+        /// ```
+        /// #![feature(integer_sign_cast)]
+        ///
+        #[doc = concat!("let n = ", stringify!($SelfT), "::MAX;")]
+        ///
+        #[doc = concat!("assert_eq!(n.cast_signed(), -1", stringify!($SignedT), ");")]
+        /// ```
+        #[unstable(feature = "integer_sign_cast", issue = "125882")]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline(always)]
+        pub const fn cast_signed(self) -> $SignedT {
+            self as $SignedT
+        }
+
         /// Shifts the bits to the left by a specified amount, `n`,
         /// wrapping the truncated bits to the end of the resulting integer.
         ///

